File: sample/src/main/java/com/sha/formvalidatorsample/ui/RxFormActivity.java
package com.sha.formvalidatorsample.ui;

import android.app.Activity;
import android.os.Bundle;
import android.widget.Toast;

import com.sha.formvalidator.FormValidator;
import com.sha.formvalidator.RxFormValidator;
import com.sha.formvalidator.widget.FormEditText;
import com.sha.formvalidatorsample.R;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class RxFormActivity extends Activity {

    private FormEditText etAllowEmpty;
    private FormEditText etAlpha;
    private FormEditText etPhone;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_form);
        setupViews();
        validateOnClick();
        justCheck();
    }

    private void setupViews() {
        etAllowEmpty = findViewById(R.id.etAllowEmpty);
        etAlpha = findViewById(R.id.etAlpha);
        etPhone = findViewById(R.id.etPhone);
    }

    private void validateOnClick() {
        Disposable disposable =  new RxFormValidator<>(etAllowEmpty, etAlpha, etPhone)
                .validateOnClick(findViewById(R.id.btnValidateOnClick))
                .subscribe(isValid -> toast("Validate on Click result: " + isValid));

        compositeDisposable.add(disposable);
    }

    private void justCheck() {
        findViewById(R.id.btnJustCheck).setOnClickListener(__ -> {

            Disposable disposable = new RxFormValidator<>(etAllowEmpty, etAlpha, etPhone)
                    .validate()
                    .subscribe(isValid -> toast("Just check result: " + isValid));

            compositeDisposable.add(disposable);
        });
    }

    private void toast(String text) {
        Toast.makeText(this, text, Toast.LENGTH_SHORT)
                .show();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (compositeDisposable != null)
            compositeDisposable.dispose();
    }
}

File: sample/src/main/java/com/sha/formvalidatorsample/presentation/RxFormActivity.kt
package com.sha.formvalidatorsample.presentation

import android.app.Activity
import android.os.Bundle
import android.widget.Toast
import com.sha.formvalidatorsample.R
import io.reactivex.disposables.CompositeDisposable
import kotlinx.android.synthetic.main.activity_form.*

class RxFormActivity : Activity() {

    private val compositeDisposable = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_form)

        form.validateOnClick(btnValidateFormLayout) {
            Toast.makeText(this, "Form result: $it", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        compositeDisposable.dispose()
    }
}
